File: Chimera.LUA/namespaces/permashowmanager.lua
---@class permashowmanager
local permashowmanager = { }

---@param menu menu
---@param pos vec2
---@return permashow
function permashowmanager.add(menu, pos) end

---@param name string
---@return permashow
function permashowmanager.get(name) end

---@param permashow permashow
---@return boolean
function permashowmanager.remove(permashow) end

---@type permashowmanager
_G.permashow_manager = { }

File: Chimera.Core/Core/leaguescript/Impl/Managers/PermashowManager/PermashowManager.cpp
#include "PermashowManager.h"

PermashowManager* g_permashow_manager = new PermashowManager();

void PermashowManager::on_draw()
{
	for (const auto& permashow : permashows)
		permashow->on_draw();
}

Permashow* PermashowManager::add_permashow(Menu* menu, const Vec2& position)
{
	Permashow* permashow = new Permashow{ menu, position };
	permashows.push_back(permashow);

	return permashow;
}

Permashow* PermashowManager::get_permashow(const std::string& name)
{
	for (const auto& p : permashows)
	{
		if (p->get_name() == name)
			return p;
	}

	return nullptr;
}

bool PermashowManager::remove_permashow(Permashow* permashow)
{
	return false;
}

void PermashowManager::remove_permashows()
{
	for (const auto& elt : permashows)
		delete elt;

	permashows.clear();
}